rts/motoko-rts/src/alloc.rs

```diff
@@ -11,14 +11,14 @@ pub use alloc_impl::alloc_words;
 #[cfg(feature = "gc")]
 pub(crate) use alloc_impl::grow_memory;
 
+use crate::constants::WASM_HEAP_SIZE;
 use crate::rts_trap_with;
 use crate::types::{size_of, Array, Blob, Bytes, SkewedPtr, Words, TAG_ARRAY, TAG_BLOB};
 
 #[no_mangle]
 pub unsafe extern "C" fn alloc_array(len: u32) -> SkewedPtr {
     // Array payload should not be larger than half of the memory
-    if len > 1 << (32 - 2 - 1) {
-        // 2 for word size, 1 to divide by two
+    if Words(len) > WASM_HEAP_SIZE / 2 {
```

Comment:
Nit (feel free to ignore): how about `WASM_HEAP_SIZE_IN_WORDS` to make it clear for the reader?

Reply:
I used the type `Words`, which is a newtype and requires explicit unwrapping or conversion to `Bytes`. Do you still think `_IN_WORDS` suffix would be helpful? I can add it, just wanted to make sure you know that `Words` is a newtype and won't be implicitly converted to anything else and when used in arithmetic with `Bytes` or scalars will cause compile errors.